### PR TITLE
chore: `get_<dependency>` cleanup

### LINKED ISSUE #1072: chore: in `narwhals/_polars`, use `import polars as pl  # ignore-banned-import` instead of `get_polars`
same motivation as in https://github.com/narwhals-dev/narwhals/pull/788

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [x] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Closes #1072 

## Checklist

- [x] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

### CODE REVIEW

File: narwhals/_arrow/dataframe.py [hunk 1]
```diff
@@ -49,7 +48,11 @@ def __narwhals_namespace__(self) -> ArrowNamespace:
         return ArrowNamespace(backend_version=self._backend_version)
 
     def __native_namespace__(self) -> Any:
-        return get_pyarrow()
+        if self._implementation is Implementation.PYARROW:
+            return self._implementation.to_native_namespace()
+
+        msg = f"Expected pyarrow, got: {type(self._implementation)}"  # pragma: no cover
+        raise AssertionError(msg)
```
Comment (FBruzzesi) [inline narwhals/_arrow/dataframe.py:55]: Do we need this check? There was in pandas, so I added it everywhere

File: narwhals/_polars/namespace.py [hunk 2]
```diff
@@ -77,38 +79,42 @@ def concat(
...
 
     def mean(self, *column_names: str) -> PolarsExpr:
+        import polars as pl  # ignore-banned-import()
+
         from narwhals._polars.expr import PolarsExpr
 
-        pl = get_polars()
         if self._backend_version < (0, 20, 4):  # pragma: no cover
-            return PolarsExpr(pl.mean([*column_names]))
+            return PolarsExpr(pl.mean([*column_names]))  # type: ignore[arg-type]
```
Comment (FBruzzesi) [inline narwhals/_polars/namespace.py:110]: Now mypy pickes up polars module, which is good :)

File: narwhals/_polars/series.py [hunk 3]
```diff
@@ -20,14 +19,12 @@
 from narwhals._polars.utils import narwhals_to_native_dtype
 from narwhals._polars.utils import translate_dtype
 
-PL = get_polars()
```
Comment (FBruzzesi) [inline narwhals/_polars/series.py:23]: Not sure what was the reason for this

File: narwhals/_arrow/dataframe.py (hunk 1 above)
Comment (MarcoGorelli) [reply narwhals/_arrow/dataframe.py:55]: probably not needed (but doesn't hurt to have it)

File: narwhals/_arrow/series.py [hunk 4]
```diff
@@ -304,7 +302,11 @@ def n_unique(self) -> int:
         return pc.count(unique_values, mode="all")  # type: ignore[no-any-return]
 
     def __native_namespace__(self) -> Any:  # pragma: no cover
-        return get_pyarrow()
+        if self._implementation is Implementation.PYARROW:
+            return self._implementation.to_native_namespace()
```
Comment (MarcoGorelli) [inline narwhals/_arrow/series.py:306]: while we're here, could we add a test which hits this? refactors with `# pragma: no cover` are always a bit scary 😄

**Review by MarcoGorelli — COMMENTED** [review]
yay thanks!

File: narwhals/_arrow/series.py (hunk 4 above)
Comment (FBruzzesi) [reply narwhals/_arrow/series.py]: I removed the `# pragma: no cover` and shows to be covered (somewhere).
Would you prefer having dedicated tests?

Comment (MarcoGorelli) [reply narwhals/_arrow/series.py]: naah so long as it's covered it's fine 🤠

File: narwhals/_arrow/series.py [hunk 5]
```diff
@@ -573,7 +577,8 @@ def to_frame(self: Self) -> ArrowDataFrame:
         return ArrowDataFrame(df, backend_version=self._backend_version)
 
     def to_pandas(self: Self) -> Any:
-        pd = get_pandas()
+        import pandas as pd  # ignore-banned-import()
```
Comment (MarcoGorelli) [inline narwhals/_arrow/series.py:580]: ooh nice

File: narwhals/dependencies.py [hunk 6]
```diff
@@ -93,22 +93,22 @@ def is_pandas_series(ser: Any) -> TypeGuard[pd.Series[Any]]:
...
 
 
 def is_cudf_dataframe(df: Any) -> TypeGuard[cudf.DataFrame]:
     """Check whether `df` is a cudf DataFrame without importing cudf."""
-    return (pd := get_cudf()) is not None and isinstance(df, pd.DataFrame)
+    return (cudf := get_cudf()) is not None and isinstance(df, cudf.DataFrame)
 
 
-def is_cudf_series(ser: Any) -> TypeGuard[pd.Series[Any]]:
+def is_cudf_series(ser: Any) -> TypeGuard[cudf.Series[Any]]:
```
Comment (MarcoGorelli) [inline narwhals/dependencies.py:109]: 🙈

**Review by MarcoGorelli — APPROVED** [review]
legend mate, love these kinds of clean ups

**Pull request merged**